### PR TITLE
8294066: IGV: Graph changes when deleting a graph in the same group with smaller index

### PR DESCRIPTION
**Pull request by jyukutyo**
IGV displays a different graph when removing a graph that has a smaller index than the graph displayed in the window. We can reproduce this behavior with the following steps.

1. Open a graph file. Open a graph.
2. In the outline window, remove a graph that has a smaller index than the open graph.
3. The graph changes from the open graph to another graph.

With this pull request, IGV will keep the same graph when doing the above.

# Tests
There are no test classes corresponding to the classes this PR changes. So I did manual tests.

I open the graph at index 8, a diff graph that contains this graph, a graph that clones this graph, and a graph that is in a different folder.

スクリーンショット_20230125_171428

I remove the graph at index 1. IGV still shows the same charts and the index of the chart changes from 8 to 7. This affects not only a single graph, but also diff graphs and clone graphs. The graph in another folder, which is "12. Global code motion...",  is not affected.

スクリーンショット_20230125_171457

Next, I remove the graph at index 10. Its index is greater than the index of the display graph. This time, the diff graph contains the graph at index 12. So the index has changed from 12 to 11.

スクリーンショット_20230125_171528

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8294066](https://bugs.openjdk.org/browse/JDK-8294066): IGV: Graph changes when deleting a graph in the same group with smaller index

### Reviewers
 * [Roberto Castañeda Lozano](https://openjdk.org/census#rcastanedalo) (@robcasloz - **Reviewer**)
 * [Tobias Holenstein](https://openjdk.org/census#tholenstein) (@tobiasholenstein - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12210/head:pull/12210` \
`$ git checkout pull/12210`

Update a local copy of the PR: \
`$ git checkout pull/12210` \
`$ git pull https://git.openjdk.org/jdk pull/12210/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12210`

View PR using the GUI difftool: \
`$ git pr show -t 12210`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12210.diff">https://git.openjdk.org/jdk/pull/12210.diff</a>

</details>

### CODE REVIEW

**Review by robcasloz — APPROVED** [review]
Thanks for fixing this,  Koichi! Please update the copyright year and remove unused imports, otherwise it looks good to me.

**Comment by jyukutyo** [comment]
> Please update the copyright year and remove unused imports, otherwise it looks good to me.

@robcasloz  Thank you for your kind advice! I've fixed them.

**Review by robcasloz — APPROVED** [review]

File: src/utils/IdealGraphVisualizer/Coordinator/src/main/java/com/sun/hotspot/igv/coordinator/actions/GraphRemoveCookie.java [hunk 1]
```diff
@@ -34,6 +40,28 @@ public GraphRemoveCookie(InputGraph graph) {
...
+                    }
+                    int firstPosition = model.getFirstPosition();
+                    int secondPosition = model.getSecondPosition();
+                    int targetPosition = list.indexOf(graph);
+                    if (targetPosition < firstPosition) {
+                        firstPosition--;
+                    }
+                    if (targetPosition < secondPosition) {
+                        secondPosition--;
+                    }
```
Comment (tobiasholenstein) [inline src/utils/IdealGraphVisualizer/Coordinator/src/main/java/com/sun/hotspot/igv/coordinator/actions/GraphRemoveCookie.java:60]:
```suggestion
                    }
                    if (targetPosition == firstPosition || targetPosition == secondPosition) {
                        t.close();
                    }
```

**Review by tobiasholenstein — CHANGES_REQUESTED** [review]
@jyukutyo thanks for fixing this! 

One suggestion I have is to close the open "EditorTopComponent" when it displays a graph that has been deleted (see my code suggestion)

Otherwise it works like expected.

File: src/utils/IdealGraphVisualizer/Coordinator/src/main/java/com/sun/hotspot/igv/coordinator/actions/GraphRemoveCookie.java (hunk 1 above)
Comment (jyukutyo) [reply src/utils/IdealGraphVisualizer/Coordinator/src/main/java/com/sun/hotspot/igv/coordinator/actions/GraphRemoveCookie.java:64]: Thank you for your suggestion. I've put this code before decrementing.

Comment (tobiasholenstein) [reply src/utils/IdealGraphVisualizer/Coordinator/src/main/java/com/sun/hotspot/igv/coordinator/actions/GraphRemoveCookie.java:64]: sure. thanks!

**Review by tobiasholenstein — APPROVED** [review]
looks good to me!

**Pull request closed without merging**